android/src/main/java/com/reactnativecommunity/webview/URLUtil.java

```diff
@@ -135,25 +135,32 @@ public static final String guessFileName(
 
     /** Regex used to parse content-disposition headers */
     private static final Pattern CONTENT_DISPOSITION_PATTERN =
-            Pattern.compile("attachment(?:;\\s*filename\\s*=\\s*(\"?)([^\"]*)\\1)?(?:;\\s*filename\\s*\\*\\s*=\\s*([^']+)'[^']*'([^']*))?\\s*$",
+            Pattern.compile("attachment(?:;\\s*filename\\s*=\\s*(\"?)([^\"]*)\\1)?(?:;\\s*filename\\s*\\*\\s*=\\s*([^']*)'[^']*'([^']*))?\\s*$",
             Pattern.CASE_INSENSITIVE);
 
     /**
      * Parse the Content-Disposition HTTP Header. The format of the header
-     * is defined here: http://www.w3.org/Protocols/rfc2616/rfc2616-sec19.html
+     * is defined here: <a href="https://www.rfc-editor.org/rfc/rfc6266">RFC 6266</a>
      * This header provides a filename for content that is going to be
      * downloaded to the file system. We only support the attachment type.
-     * Note that RFC 2616 specifies the filename value must be double-quoted.
-     * Unfortunately some servers do not quote the value so to maintain
-     * consistent behaviour with other browsers, we allow unquoted values too.
      */
     static String parseContentDisposition(String contentDisposition) {
         try {
+            // The regex attempts to match the following pattern:
+            //     attachment; filename="(Group 2)"; filename*=(Group 3)'(lang)'(Group 4)
+            // (Group 1 refers to the quotation marks around Group 2)
+            //
+            // Test cases can be found at http://test.greenbytes.de/tech/tc2231/
```

Comment:
Thanks for including that! Seems to have other cases that are not supported, such as `filename=foo.html` (no quotes) or breaking things up with `filename*1`/`filename*2`...

But it doesn't look like your changes break support for a previously-working case, so that should be good.

Reply:
Yes, that's the case. If there remain issues in the future regarding filename guessing, I would like to switch to a more robust implementation.

Thank you!